lease-kubernetes/src/main/scala/akka/coordination/lease/kubernetes/internal/KubernetesApiImpl.scala

```diff
@@ -228,6 +227,8 @@ PUTs must contain resourceVersions. Response:
 
   private def pathForLease(name: String): Uri.Path =
     Uri.Path.Empty / "apis" / "akka.io" / "v1" / "namespaces" / namespace / "leases" / name
+      .replaceAll("[^\\d|\\w|\\-|\\.]", "")
```

Comment:
You don't need `|` characters in a regex
```suggestion
      .replaceAll("[^\\d\\w\\-\\.]", "")
```

Reply:
Oh right , I thought the or works more like an if else so is mutually exclusive and in this case should be more performant. You don't need but unless I have not understood something without it will match like a bunch of ifs

Reply:
No, in this case the `|` character is just one of the characters in the `[]`, so this would be wrong for entities with `|` characters in their name:

```
@ "asdf|.-!000".replaceAll("[^\\d|\\w|\\-|\\.]", "") 
res0: String = "asdf|.-000"
```

Reply:
Maybe it needs to be escaped but I really doubt it will make much difference at all to the performance so let's run with your last one 🙂 thanks for your time on it.